### DISCUSSION TITLE
[General] How to set up a zksync RPC node. Is there any tutorial?

### DISCUSSION THREAD

**Discussion by y3zha** [zkSync-Community-Hub/zksync-developers#343] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I am reading this [tutorial](https://docs.zksync.io/infra/configuration.html), but there is no details and I dont know what to do.

### What is the specific issue or error you're encountering?

There is no tutorial to learn.

### Can you share the error messages or logs you're receiving, if any?

/

### Have you made any recent changes to the contract before encountering this issue?

/

### Are there any external libraries or contracts that your contract interacts with?

/

### Can you provide the relevant portions of your contract code where the issue is occurring?

/

### Have you tried to isolate the problem, and if so, what were the results?

/

### What steps have you already taken to try to resolve the issue?

/

**Comment by jaguard2021** [comment]
Have you ever run node directly on your local computer before?

**Reply by N1ghtSh4de** [reply]
Not a good idea to run on a local computer. VPS is the way to go.

**Reply by jaguard2021** [reply]
Yes, using a VPS is more flexible and does not require significant resources. Running Node self requires deep knowledge and experience in managing Node

**Comment by Bouncyknighter** [comment]
I am working on Tutorial, but try out GETH

**Comment by FrondEnt** [comment]
what about Alchemy? 
[you can make your own private rpc](https://www.alchemy.com/chain-connect/chain/zksync-era-mainnet)

**Comment by dutterbutter** [comment, marked as answer]
Hey @y3zha 

There is no particular tutorial for step by step walk through on how to run an external node. We can look to put one together in the future. For now please refer to the documentation [here](https://docs.zksync.io/infra), and [here](https://github.com/matter-labs/zksync-era/tree/main/docs/guides/external-node). 

If you run into a particular issue during the setup please open a new discussion. Thanks!

**Comment by npam90** [comment]
Please provide more specific details or context about the tutorial you're referring to so that I can assist you better.

**Comment by atzipajabken** [comment]
The user is encountering an issue with a tutorial they are attempting to follow on a Testnet environment, but reports that there are no details or instructions provided. They have not encountered any specific errors or made any changes to the contract, and have not provided any additional details or repository link.
